components/brave_vpn/brave_vpn_os_connection_api.cc

```diff
@@ -402,59 +398,20 @@ void BraveVPNOSConnectionAPI::ParseAndCacheHostnames(
           << hostname_->display_name << ", " << hostname_->is_offline << ", "
           << hostname_->capacity_score;
 
-  if (skus_credential_.empty()) {
-    VLOG(2) << __func__ << " : skus_credential is empty";
-    UpdateAndNotifyConnectionStateChange(ConnectionState::CONNECT_FAILED);
-    return;
-  }
-
   if (!GetAPIRequest()) {
     CHECK_IS_TEST();
     return;
   }
 
   // Get subscriber credentials and then get EAP credentials with it to create
   // OS VPN entry.
-  VLOG(2) << __func__ << " : request subscriber credential:"
+  VLOG(2) << __func__ << " : request profile credential:"
           << GetBraveVPNPaymentsEnv(GetCurrentEnvironment());
-  GetAPIRequest()->GetSubscriberCredentialV12(
-      base::BindOnce(&BraveVPNOSConnectionAPI::OnGetSubscriberCredentialV12,
-                     base::Unretained(this)),
-      skus_credential_, GetBraveVPNPaymentsEnv(GetCurrentEnvironment()));
-}
-
-void BraveVPNOSConnectionAPI::OnGetSubscriberCredentialV12(
-    const std::string& subscriber_credential,
-    bool success) {
-  if (cancel_connecting_) {
-    UpdateAndNotifyConnectionStateChange(ConnectionState::DISCONNECTED);
-    cancel_connecting_ = false;
-    return;
-  }
-
-  if (!success) {
-    VLOG(2) << __func__ << " : failed to get subscriber credential";
-    if (subscriber_credential == kTokenNoLongerValid) {
-      for (auto& obs : observers_)
-        obs.OnGetInvalidToken();
-    }
-    UpdateAndNotifyConnectionStateChange(ConnectionState::CONNECT_FAILED);
-    return;
-  }
-
-  VLOG(2) << __func__ << " : received subscriber credential";
-
-  if (!GetAPIRequest()) {
-    CHECK_IS_TEST();
-    return;
-  }
 
-  // TODO(bsclifton): consider storing `subscriber_credential` for
-  // support ticket use-case (see `CreateSupportTicket`).
   GetAPIRequest()->GetProfileCredentials(
       base::BindOnce(&BraveVPNOSConnectionAPI::OnGetProfileCredentials,
                      base::Unretained(this)),
-      subscriber_credential, hostname_->hostname);
+      GetSubscriberCredential(local_prefs_), hostname_->hostname);
```

Comment:
If `GetSubscriberCredential` returns `""` due to being expired, we would need to know how to refetch the credential

Reply:
When testing, it would be hard to make this happen. But would happen if person had Brave open for a while after `CheckInitialState` is called (like if they had open for multiple days)

Reply:
sounds good to me

Reply:
> @simonhong on android, we call `ReloadPurchasedState()` on each app open to check if the subscription is still valid.. are we fetching new credential on `ReloadPurchasedState` ?

So far, new skus_credential is fetched whenever `ReloadPurchasedState()` and new subscriber credential is fetched when creating new vpn entry on desktop. With this PR, skus credential and subscriber credential are fetched only when it's expired by storing subscriber credential in local state.

Reply:
With background timer that will launch right before the expired time, subscriber credential will be refreshed.